Security/Authentication/Token/OAuth2Token.php

```diff
@@ -39,7 +40,7 @@ private function buildRolesFromScopes(): array
         $roles = [];
 
         foreach ($this->getAttribute('server_request')->getAttribute('oauth_scopes', []) as $scope) {
-            $roles[] = sprintf('ROLE_OAUTH2_%s', trim(strtoupper($scope)));
+            $roles[] = $this->getAttribute('role_prefix') . trim(strtoupper($scope));
```

Comment:
@Allypost You should ensure that the entire role is now uppercase since now we're giving the developer an option to customize the prefix.

Reply:
@spideyfusion Should be fixed now. The role should now also be trimmed as a whole instead of just trimming the `scope` and leaving the `role_prefix` filthy with whitespace.